### PR TITLE
Simplify loss extraction to loss.item()

### PR DESCRIPTION
**Pull request by bkowshik**
## What

One-line cleanup in the training loop:

```diff
-            total_loss = loss.cpu().detach().item()
+            total_loss = loss.item()
```

## Why

`.item()` on a scalar tensor already does the whole job: it detaches from the autograd graph, copies from the device to host memory, and converts to a Python number — on any device. The explicit `.cpu()` and `.detach()` calls add nothing.

Behavior is identical — `total_loss` is the same Python float either way.

### CODE REVIEW

**Comment by AlexanderPfefferle** [comment]
Another good catch!
